src/components/FileRepo/DownloadButton.tsx

```diff
@@ -0,0 +1,216 @@
+/* eslint-disable react/prop-types */
+import React, { Fragment, FunctionComponent } from 'react';
+// @ts-ignore
+import { compose } from 'recompose';
+// @ts-ignore
+import { injectState } from 'freactal';
+import uniq from 'lodash/uniq';
+// @ts-ignore
+import { ColumnsState } from '@kfarranger/components/dist/DataTable';
+import { familyMemberAndParticipantIds } from '../FamilyManifestModal';
+import { TRACKING_EVENTS, trackUserInteraction } from 'services/analyticsTracking';
+import {
+  clinicalDataFamily,
+  clinicalDataParticipants,
+  downloadBiospecimen,
+} from 'services/downloadData';
+import { withApi } from 'services/api';
+import FamilyManifestModal from '../FamilyManifestModal/FamilyManifestModal';
+
+import './DownloadButton.css';
+import { Menu, message, Spin } from 'antd';
+import ReportsButton from 'ui/ReportsButton';
+import { Sqon, ModalStateEffects } from '../../types';
+
+type Column = {
+  Header: string;
+  accessor: string;
+  canChangeShow: boolean;
+  extendedDisplayValues: any;
+  extendedType: string;
+  field: string;
+  jsonPath: any;
+  query: any;
+  show: boolean;
+  sortable: boolean;
+  type: string;
+  [key: string]: any;
+};
+
+type DownloaderParams = {
+  api?: Function;
+  sqon: Sqon;
+  columns: Column[];
+};
+
+type StateProp = {
+  type: string;
+  keyField: string;
+  defaultSorted: any;
+  columns: Column[];
+  [key: string]: any;
+};
+
+type Props = {
+  sqon: Sqon;
+  api: Function;
+  projectId: string;
+  effects: ModalStateEffects;
+};
+
+type ColumnStateRenderProps = {
+  state: StateProp;
+  loading: boolean;
+};
+
+const trackDownload = async (label: string) =>
+  // @ts-ignore
+  await trackUserInteraction({
+    category: TRACKING_EVENTS.categories.fileRepo.actionsSidebar,
+    action: TRACKING_EVENTS.actions.download.report,
+    label,
+  });
+
+const participantDownloader = async ({ api, sqon, columns }: DownloaderParams) => {
+  await trackDownload('Clinical (Participant)');
+  const isFileRepo = true;
+  const { participantIds } = await familyMemberAndParticipantIds({
+    api,
+    sqon,
+    isFileRepo,
+  });
+  const downloadConfig = {
+    sqon: {
+      op: 'in',
+      content: {
+        field: 'participants.kf_id',
+        value: participantIds,
+      },
+    },
+    columns,
+    isFileRepo: isFileRepo,
+  };
+  return clinicalDataParticipants(downloadConfig);
+};
+
+const familyDownloader = async ({ api, sqon, columns }: DownloaderParams) => {
+  await trackDownload('Clinical (Participant and family)');
+  const isFileRepo = true;
+  const { familyMemberIds, participantIds } = await familyMemberAndParticipantIds({
```

Comment:
how was the api handle before, don't we have a reducer/actions for this business logic?

Reply:
What I mean is that business logic about fetching data should not be bind the the UI component, but normally to a action/reducer/store

Was it in the same file before?

Reply:
Good point. Yes, it was in the same file. I'll keep that way for now. But, There s another ticket coming up where I might have to change these actions. I'll take that opportunity to unbind that logic